streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -891,6 +821,9 @@ private void initializeTopology() {
     }
 
     private void closeTopology(final boolean clean) {
+        if (state() != State.RUNNING) {
```

Comment:
I could see we are trying to maintain the same behavior, but still why a restoring task won't need to close topology?

Reply:
Because the topology is only initialized when restoring is finished.

Reply:
Sg, should we move the log on L811 inside the if statement?